### PR TITLE
Fix aten::any.dim truthiness for numeric tensors

### PR DESCRIPTION
**Pull request by fallintoplace**
## Summary
- Convert numeric `aten::any.dim` inputs into an explicit nonzero mask before reducing.
- Keep bool inputs on the existing bool-to-int reduction path.
- Add deterministic regression tests for cancelling numeric values and negative nonzero values.

## Testing
- `git diff --check`
- `bazelisk test //tests/core/conversion/converters:test_reduce` was attempted locally. The first run was blocked by missing PyTorch; after installing PyTorch in a local venv and setting `TORCH_PATH`, Bazel stayed in analysis for several minutes on this local machine and was interrupted before running tests.

### CODE REVIEW

**Comment by zewenli98** [comment]
@fallintoplace Thanks for the contribution. Torchscript is being removed soon (like pytorch 2.14). Please use Dynamo instead.

**Comment by fallintoplace** [comment]
@zewenli98 What about this one? Thank you for your attention.

File: core/conversion/converters/impl/reduce.cpp [hunk 1]
```diff
@@ -27,6 +27,21 @@ nvinfer1::ITensor* anyDimImplementation(
   // Reduce does not work on bool inputs
   if (in_tensor->getType() == nvinfer1::DataType::kBOOL) {
     in_tensor = castITensor(ctx, in_tensor, nvinfer1::DataType::kINT32, (util::node_info(n) + "_in").c_str());
+  } else {
+    // Numeric truthiness is based on nonzero elements, not the reduced sum of raw values.
+    auto zero_tensor = tensor_to_const(ctx, torch::tensor({0}, torch::kInt32), util::node_info(n) + "_zero");
```
Comment (zewenli98) [inline core/conversion/converters/impl/reduce.cpp]: zero_tensor with torch::kInt32 is created here and then compared with in_tensor. If in_tensor is float/half or lives on a different device, the elementwise EQUAL may fail at network construction or behave incorrectly. 
Can you create the zero constant to match in_tensor's dtype and device?

File: tests/core/conversion/converters/test_reduce.cpp [hunk 2]
```diff
@@ -327,6 +327,39 @@ TEST(Converters, ATenAnyDimAllFalseConvertsCorrectly) {
...
+  const auto graph = R"IR(
+    graph(%0 : Tensor):
+      %1 : int = prim::Constant[value=1]()
+      %3 : bool = prim::Constant[value=0]()
+      %5 : Tensor = aten::any(%0, %1, %3)
+      return (%5))IR";
+  auto in = at::tensor({{1, -1}, {0, 0}}, at::TensorOptions().dtype(at::kInt).device(at::kCUDA));
+  test_body(graph, in);
+}
+
```
Comment (zewenli98) [inline tests/core/conversion/converters/test_reduce.cpp:359]: Can you test different dtypes like fp32 and fp16? 
Can you also test if NaN works? e.g.:
```
>>> torch.any(torch.tensor([0, torch.nan]))
tensor(True)
```

**Review by zewenli98 — COMMENTED** [review]
> @fallintoplace Thanks for the contribution. Torchscript is being removed soon (like pytorch 2.14). Please use Dynamo instead.

Pytorch decided not to remove Torchscript support in 2.14.

This PR fixes the correctness bug in aten::any on numeric tensors (cases like [1, -1] that cancel to 0). Can you fix the comments and rebase? Thanks!

**Review by zewenli98 — APPROVED** [review]
LGTM

**Comment by zewenli98** [comment]
@fallintoplace Can you fix the lint issue? please refer to https://github.com/pytorch/TensorRT/blob/main/CONTRIBUTING.md#coding-guidelines. Thanks

**Comment by zewenli98** [comment]
@fallintoplace Some CI errors were fixed last week. can you rebase again? thanks!
